core/packages/PackageInfo.h

```diff
@@ -64,5 +64,25 @@ class PackageInfo {
 
     static bool lexCmp(const std::vector<core::NameRef> &lhs, const std::vector<core::NameRef> &rhs);
 };
+
+// Information about the imports of a package. The imports are split into two categories, packages whose name falls
+// within the namespace of `package`, and everything else. The reason for pre-processing the imports this way is that it
+// simplifies some work when stubbing constants for rbi generation.
+class ImportInfo final {
+public:
+    // The mangled name of the package whose imports are described.
+    core::NameRef package;
+
+    // Imported packages whose name is a prefix of `package`. For example, if the package `Foo::Bar` imports `Foo` that
+    // package's name would be in `parentImports` because its name is a prefix of `Foo::Bar`.
+    std::vector<core::NameRef> parentImports;
```

Comment:
The current code records packages that are parents of the particular package we're generating RBIs for; this only records imported parent packages...which seems to me to be a potentially smaller set.  Why do we not record general parents anymore?

(I guess maybe I am not as familiar with packages as I should be, but my naive assumption is that parent packages are automatically in scope?)

Reply:
Parent packages are only in scope if they're imported, so the current behavior is incorrect.